ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/KafkaConfigStoreTest.java

```diff
@@ -0,0 +1,204 @@
+import org.mockito.junit.MockitoRule;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+public class KafkaConfigStoreTest {
+  private final KsqlConfig ksqlConfig = new KsqlConfig(
+      ImmutableMap.of(
+        KsqlConfig.KSQL_SERVICE_ID_CONFIG, "test-service-id"
+      )
+  );
+  private final String topicName = KsqlInternalTopicUtils.getTopicName(ksqlConfig, "configs");
+  private final TopicPartition topicPartition = new TopicPartition(topicName, 0);
+  private final List<TopicPartition> topicPartitionAsList
+      = Collections.singletonList(topicPartition);
+
+  @Mock
+  private Supplier<KafkaConsumer<String, KsqlProperties>> consumerSupplier;
+  @Mock
+  private Supplier<KafkaProducer<String, KsqlProperties>> producerSupplier;
+  @Mock
+  private KafkaProducer<String, KsqlProperties> producer;
+  @Mock
+  private KafkaTopicClient topicClient;
+
+  private KafkaConfigStore configStore;
+
+  @Rule
+  public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
+  @Before
+  @SuppressWarnings("unchecked")
+  public void setUp() {
+    when(topicClient.isTopicExists(anyString())).thenReturn(true);
+    when(topicClient.addTopicConfig(anyString(), anyMap()))
+        .thenReturn(false);
+    when(producerSupplier.get()).thenReturn(producer);
+  }
+
+  private void initConfigStore() {
+    configStore = new KafkaConfigStore(ksqlConfig, topicClient, consumerSupplier, producerSupplier);
+  }
+
+  @SuppressWarnings("unchecked")
+  private KafkaConsumer<String, KsqlProperties> expectRead(
+      final KafkaConfigStore.KsqlProperties... properties) {
+    final KafkaConsumer<String, KsqlProperties> consumer = mock(KafkaConsumer.class);
+    when(consumer.endOffsets(any()))
+        .thenReturn(ImmutableMap.of(topicPartition, (long)properties.length));
+    when(consumer.position(any())).thenReturn(0L).thenReturn((long) properties.length);
+    final List<ConsumerRecord<String, KafkaConfigStore.KsqlProperties>> records
+        = new LinkedList<>();
+    for (int i = 0; i < properties.length; i++) {
+      records.add(
+          new ConsumerRecord<>(
+              topicName,
+              0,
+              (long) i,
+              "ksql-standalone-configs",
+              properties[i]
+          )
+      );
+    }
+    when(consumer.poll(any())).thenReturn(
+        new ConsumerRecords<>(
+            Collections.singletonMap(
+                topicPartition,
+                records
+            )
+        )
+    );
+    return consumer;
+  }
+
+  private void verifyDrainLog(
+      final KafkaConsumer<String, KsqlProperties> consumer,
+      final InOrder inOrder,
+      final int nmsgs) {
+    inOrder.verify(consumer).assign(topicPartitionAsList);
+    inOrder.verify(consumer).seekToBeginning(topicPartitionAsList);
+    for (int i = 0; i < nmsgs; i++) {
+      inOrder.verify(consumer).poll(any());
+    }
+  }
+
+  @Test
+  public void shouldWriteConfigIfNoConfigWritten() {
```

Comment:
This test would benefit from being split to make what's tesitng more explicit:

- shouldWriteConfigIfNoConfigWritten: should only test it writes current config if no existing:

```java
// When:
configStore.getKsqlConfig()

// Then:
verify(producer).produce(blah);
```

- shouldReadConfigAfterWrite: tests it reads after the write to ensure consistency:
```
// Given:
givenSecondConsumerReturns(expectedConfig);

// When:
final KsqlConfig resolvedConfig = configStore.getKsqlConfig();

// Then:
assertThat(resolvedConfig, is(expectedConfig));
```